### ISSUE TITLE
support coloresque for css language

### ISSUE THREAD

**Issue by marslo** [sharkdp/bat#2808]
I'm using vim plugin [gko/vim-coloresque](https://github.com/gko/vim-coloresque) to show css coloresque ( as below screenshot ).

![vim-coloresque-css](https://github.com/sharkdp/bat/assets/5822057/eef88e78-799e-41ac-8148-893b65134be2)

Is it feasible to develop such a feature?

**Comment by eth-p** [comment]
This isn't possible to do for CSS specifically, but it would be possible to do in a more generic way by postprocessing the output. I think this would be a nice feature to have, but it would likely be opt-in with an argument to enable it.

My main concerns with having it enabled by default are:
- Not all users are using a terminal capable of 24-bit color. 
- Not all users may want to use it.
- Adding more postprocessing may have an impact on performance, especially if we have to use color quantization to support terminals that only have a 256-color palette.

Thoughts, @sharkdp, @keith-hall, @Enselic?

**Comment by Enselic** [comment]
Hmm, I don't understand. We already support 24-bit colors. What would "css coloresque" give us that we don't have?

**Comment by eth-p** [comment]
@Enselic Apologies for the clarity on my comment.

The original request by marslo is to have `bat` highlight hex color triplets such as `#ffceb0` in such a way that it has the same background color as the triplet itself. For example, the above color would be rendered in the terminal like 
<img width="178" alt="image" src="https://github.com/sharkdp/bat/assets/32112321/37008ca0-af90-423c-a8fe-c59f1c892ba0">.

My concern with adding support for this is that not all terminals support the sequence for 24-bit color. If we were to add the feature, we would probably need code to reduce the full color down to one of the 256 paletted colors.

**Comment by Enselic** [comment]
No worries. But can't this be solved purely with syntax highlighting? I don't think we need post processing?

**Comment by keith-hall** [comment]
Syntax highlighting can only apply scopes to certain text which colour schemes can target. To do this purely in syntax highlighting without post processing would mean the syntax definition would need to create a unique scope for each possible hex color, and then the colour scheme/theme in use would have to target each color scope and assign it the same color as specified in the scope... 
TL;DR: post processing is a much more realistic option

**Comment by sharkdp** [comment]
We even had an implementation for this here: https://github.com/sharkdp/bat/pull/2782

I closed it recently. Feel free to further discuss it here if you think it would be a good fit for bat after all.

**Comment by marslo** [comment]
Hi @sharkdp , seems the push request https://github.com/sharkdp/bat/pull/2782/commits haven't been pushed into `master` branch. I didn't get any clue on params `--render-hex-colors`

details:
```bash
$ bat fzf-color-theme.css -- --render-hex-colors
       File: fzf-color-theme.css
   1   /**
   2     bg+        : #3F3F3F,
   3     bg         : #4B4B4B,
   4     border     : #6B6B6B,
   5     spinner    : #98BC99,
   6     hl         : #719872,
   7     fg         : #D9D9D9,
   8     header     : #719872,
   9     info       : #BDBB72,
  10     pointer    : #E12672,
  11     marker     : #E17899,
  12     fg+        : #D9D9D9,
  13     preview-bg : #3F3F3F,
  14     prompt     : #98BEDE,
  15     hl+        : #98BC99,
  16   **/
[bat error]: '--render-hex-colors': No such file or directory (os error 2)
```

### diagnostic
$ bat fzf-color-theme.css --diagnostic
#### Software version

bat 0.24.0 (fc95468)

#### Operating system

Linux 5.15.133.1-microsoft-standard-WSL2

#### Command-line

```bash
bat fzf-color-theme.css --diagnostic
```

#### Environment variables

```bash
SHELL=/bin/bash
PAGER=<not set>
LESS=-eirMXR
LANG=en_US.UTF-8
LC_ALL=en_US.UTF-8
BAT_PAGER='less -RF'
BAT_PAGING=<not set>
BAT_CACHE_PATH=<not set>
BAT_CONFIG_PATH=<not set>
BAT_OPTS=<not set>
BAT_STYLE=<not set>
BAT_TABS=<not set>
BAT_THEME=gruvbox-dark
XDG_CONFIG_HOME=<not set>
XDG_CACHE_HOME=<not set>
COLORTERM=<not set>
NO_COLOR=<not set>
MANPAGER='bash -c '\''col -bx | bat -l man --style plain --paging always'\'''
```

#### System Config file

Could not read contents of '/etc/bat/config': No such file or directory (os error 2).

#### Config file

```
# This is `bat`s configuration file. Each line either contains a comment or
# a command-line option that you want to pass to `bat` by default. You can
# run `bat --help` to get a list of all possible configuration options.

# Specify desired highlighting theme (e.g. "TwoDark"). Run `bat --list-themes`
# for a list of all available themes
--theme="gruvbox-dark"
--style="numbers,changes,header"

# Enable this to use italic text on the terminal. This is not supported on all
# terminal emulators (like tmux, by default):
--italic-text=always

# Uncomment the following line to disable automatic paging:
#--paging=never

# Uncomment the following line if you are using less version >= 551 and want to
# enable mouse scrolling support in `bat` when running inside tmux. This might
# disable text selection, unless you press shift.
--pager="less --RAW-CONTROL-CHARS --quit-if-one-screen --mouse"

# Syntax mappings: map a certain filename pattern to a language.
#   Example 1: use the C++ syntax for Arduino .ino files
#   Example 2: Use ".gitignore"-style highlighting for ".ignore" files
--map-syntax "*.ino:C++"
--map-syntax ".ignore:Git Ignore"
--map-syntax='*.conf:INI'
--map-syntax='/etc/apache2/**/*.conf:Apache Conf'
--map-syntax "**/jenkinsfile/**/*:Groovy"
```

#### Custom assets metadata

Could not read contents of '/home/marslo/.cache/bat/metadata.yaml': No such file or directory (os error 2).

#### Custom assets

'/home/marslo/.cache/bat' not found

#### Compile time information

- Profile: release
- Target triple: x86_64-unknown-linux-musl
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-unknown-linux-gnu

#### Less version

```
> less --version
less 590 (GNU regular expressions)
Copyright (C) 1984-2021  Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Home page: https://greenwoodsoftware.com/less
```

**Comment by marslo** [comment]
it's me again :) I created a tool for color preview - [batchip](https://github.com/marslo/batchip), fully support bat

![batchip](https://github.com/marslo/batcolor/raw/main/screenshots/batchip.png)
